Refactor API methods to remove unnecessary async/await and improve return handling

Oksamies opened the pull request:
Refactor API methods to remove unnecessary async/await and improve return handling

Note: Some of the functions return shapes differ as of this commit, so the places that use these functions should change in future commits

Update dapper-fake functions to match the changed dapper methods return shapes (pagination)

anttimaki reviewed (commented):
Nothing to comment here except that the [comment from previous PR's review](https://github.com/thunderstore-io/thunderstore-ui/pull/1614#discussion_r2570750682) applies here too.

Oksamies committed:
Refactor API methods to remove unnecessary async/await and improve return handling

Note: Some of the functions return shapes differ as of this commit, so the places that use these functions should change in future commits

Oksamies committed:
Update dapper-fake functions to match the changed dapper methods return shapes (pagination)

Oksamies committed:
Fix import statement to correctly specify type for RatedPackagesResponseData

Oksamies commented:
Closing as too stale
